sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/testresource/StargateTestResource.java

```diff
@@ -0,0 +1,279 @@
+
+    cassandraContainer =
+        new GenericContainer<>(getCassandraImage())
+            .withEnv("HEAP_NEWSIZE", "512M")
+            .withEnv("MAX_HEAP_SIZE", "2048M")
+            .withEnv("CASSANDRA_CGROUP_MEMORY_LIMIT", "true")
+            .withEnv(
+                "JVM_EXTRA_OPTS",
+                "-Dcassandra.skip_wait_for_gossip_to_settle=0 -Dcassandra.load_ring_state=false -Dcassandra.initial_token=1")
+            .withNetworkAliases("cassandra")
+            .withExposedPorts(7000)
+            .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("cassandra-docker")))
+            .waitingFor(Wait.forLogMessage(".*Created default superuser role.*\\n", 1))
+            .withStartupTimeout(Duration.ofMinutes(2))
+            .withReuse(reuse);
+    if (isDse()) {
+      cassandraContainer.withEnv("CLUSTER_NAME", getClusterName()).withEnv("DS_LICENSE", "accept");
+
+    } else {
+      cassandraContainer.withEnv("CASSANDRA_CLUSTER_NAME", getClusterName());
+    }
+    if (containerNetworkId.isPresent()) {
+      cassandraContainer.withNetworkMode(containerNetworkId.get());
+    } else {
+      cassandraContainer.withNetwork(network());
+    }
+    // resolve the network based on the mode
+    cassandraContainer.start();
+
+    stargateContainer =
+        new GenericContainer<>(getStargateImage())
+            .withEnv("JAVA_OPTS", "-Xmx1G")
+            .withEnv("CLUSTER_NAME", getClusterName())
+            .withEnv("CLUSTER_VERSION", getClusterVersion())
+            .withEnv("SEED", "cassandra")
+            .withEnv("SIMPLE_SNITCH", "true")
+            .withEnv("ENABLE_AUTH", "true")
+            .withNetworkAliases("coordinator")
+            .withExposedPorts(8091, 8081, 8084)
+            .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("coordinator-docker")))
+            .waitingFor(Wait.forHttp("/checker/readiness").forPort(8084).forStatusCode(200))
+            .withStartupTimeout(Duration.ofMinutes(2))
+            .withReuse(reuse);
+    // enable DSE if needed
+    if (isDse()) {
+      stargateContainer.withEnv("DSE", "1");
+    }
+    // resolve the network based on the mode
+    if (containerNetworkId.isPresent()) {
+      stargateContainer.withNetworkMode(containerNetworkId.get());
+    } else {
+      stargateContainer.withNetwork(network());
+    }
+    stargateContainer.start();
+
+    Integer authPort = stargateContainer.getMappedPort(8081);
+    Integer bridgePort = stargateContainer.getMappedPort(8091);
+
+    // get auth token
+    String token = getAuthToken("localhost", authPort);
+    LOG.info("Using auth token %s for integration tests.".formatted(token));
+
+    // return a map containing the configuration the application needs to use the service
+    ImmutableMap.Builder<String, String> propsBuilder = ImmutableMap.builder();
+    propsBuilder.put("stargate.auth.token-resolver.type", "fixed");
+    propsBuilder.put("stargate.auth.token-resolver.fixed.token", token);
```

Comment:
Yes so I will leave this open.. If we don't want to have end-to-end tests, then we can improve by injecting the auth token and use it in the tests instead of a dummy value. Adding TODO..